### ISSUE TITLE
No upload possible, OAuth error

### ISSUE THREAD

**Issue by gbomacfly** [Chocobozzz/PeerTube#4137]
Hi there,

I've managed to install Peertube on a Container under Proxmox.
I'm managing my webserver/SSL under a windows-system with IIS, so i dont need https in the container, and i removed all ssl configs in the nginx-config.

[nginx.conf.txt](https://github.com/Chocobozzz/PeerTube/files/6566867/nginx.conf.txt)

The peertube config:

[peertube.conf.txt](https://github.com/Chocobozzz/PeerTube/files/6566870/peertube.conf.txt)

Every reload shows this error:
>Fehler
Referenzen des OAuth-Clients können nicht abgerufen werden: undefined. Stellen Sie sicher, dass PeerTube korrekt konfiguriert ist (Ordner config/), speziell der Abschnitt "webserver".

Importing through i.e. youtube works fine. Uploading a file not. root or normal user. Hang at 0.0x%, after a while ist says "Unknow Error"
/admin/system/debug shows the ip from my webserver, not my public ip, so I tried to alter `trust_proxy`:

>trust_proxy:
>  \- 'loopback'`
> \- '192.168.178.0/24'

I tried with `192.168.178.0/24`, `192.168.178.0/16`, `webserverhostname`, I also tried different syntax:

>trust_proxy:
>  \- 'loopback', '192.168.178.0/24'

But this gives me an error

>May 30 21:15:49 gbosrv-peertube peertube[26979]:     at internal/main/run_main_module.js:17:47 {
May 30 21:15:49 gbosrv-peertube peertube[26979]:   "error": {
May 30 21:15:49 gbosrv-peertube peertube[26979]:     "stack": "TypeError: invalid IP address: loopback, 192.168.178.0\n    at parseipNotation (/var/www/peertube/versions/peertube-v3.2.0/node_modules/proxy-addr/index.js:163:11)\n    at compileRangeSubnets (/var/www/peertube/versions/peertube-v3.2.0/node_modules/proxy-addr/index.js:126:23)\n    at Function.compile (/var/www/peertube/versions/peertube-v3.2.0/node_modules/proxy-addr/index.js:112:23)\n    at exports.compileTrust (/var/www/peertube/versions/peertube-v3.2.0/node_modules/express/lib/utils.js:238:20)\n    at Function.set (/var/www/peertube/versions/peertube-v3.2.0/node_modules/express/lib/application.js:372:34)\n    at Object.<anonymous> (/var/www/peertube/versions/peertube-v3.2.0/dist/server.js:41:5)\n    at Module._compile (internal/modules/cjs/loader.js:999:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)\n    at Module.load (internal/modules/cjs/loader.js:863:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:708:14)\n    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)\n    at internal/main/run_main_module.js:17:47",
May 30 21:15:49 gbosrv-peertube peertube[26979]:     "message": "invalid IP address: loopback, 192.168.178.0"
May 30 21:15:49 gbosrv-peertube peertube[26979]:   },   
May 30 21:15:49 gbosrv-peertube peertube[26979]:   "stack": "TypeError: invalid IP address: loopback, 192.168.178.13\n    at parseipNotation (/var/www/peertube/versions/peertube-v3.2.0/node_modules/proxy-addr/index.js:163:11)\n    at compileRangeSubnets (/var/www/peertube/versions/peertube-v3.2.0/node_modules/proxy-addr/index.js:126:23)\n    at Function.compile (/var/www/peertube/versions/peertube-v3.2.0/node_modules/proxy-addr/index.js:112:23)\n    at exports.compileTrust (/var/www/peertube/versions/peertube-v3.2.0/node_modules/express/lib/utils.js:238:20)\n    at Function.set (/var/www/peertube/versions/peertube-v3.2.0/node_modules/express/lib/application.js:372:34)\n    at Object.<anonymous> (/var/www/peertube/versions/peertube-v3.2.0/dist/server.js:41:5)\n    at Module._compile (internal/modules/cjs/loader.js:999:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)\n    at Module.load (internal/modules/cjs/loader.js:863:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:708:14)\n    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)\n    at internal/main/run_main_module.js:17:47",

But when I try:

>trust_proxy:
>  \- 'loopback'`
> \- '192.168.178.13'

It shows the right IP, but when logged out, no user can't login again. In the log is an error:

>May 30 21:26:00 gbosrv-peertube peertube[730]: [tube.gboslave.de:443] 2021-05-30 21:26:00.529 error: Error in controller. {
May 30 21:26:00 gbosrv-peertube peertube[730]:   "err": "Error: ipaddr: the address has neither IPv6 nor IPv4 format\n    at Object.ipaddr.parse (/var/www/peertube/versions/peertube-v3.2.0/node_modules/ipaddr.js/lib/ipaddr.js:863:19)\n    at Object.isSignupAllowedForCurrentIP (/var/www/peertube/versions/peertube-v3.2.0/dist/server/helpers/signup.js:25:25)\n    at Object.<anonymous> (/var/www/peertube/versions/peertube-v3.2.0/dist/server/lib/config.js:22:46)\n    at Generator.next (<anonymous>)\n    at fulfilled (/var/www/peertube/versions/peertube-v3.2.0/node_modules/tslib/tslib.js:114:62)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
May 30 21:26:00 gbosrv-peertube peertube[730]: }

I also tried different variations of the webserver part in the peertube.conf:
>webserver:
  https: true/false
  hostname: 'mytube.domain.de'
  port: 443/80

What I tried next in the nginx config:
`proxy_set_header   Host            $host;` change to `proxy_set_header   Host            $host:443;`
but also not working.

What do I wrong? Where is my error? Can you help me?

`nginx/1.18.0 (Ubuntu)`
`npm 6.14.12`
`nodejs v12.22.11`
`ffmpeg version 4.2.4-1ubuntu0.1`

Sorry for not pasting my configs, but they break the markdown in here.

**Comment by Chocobozzz** [comment]
```
  trust_proxy:
    - 'loopback'`
    - '192.168.178.13'
```

Is the appropriate config format. It's weird that you correctly see your IP in the admin debug page, but the signup check cannot parse it :thinking: 

Could you add a debug log in `/var/www/peertube/versions/peertube-v3.2.0/dist/server/helpers/signup.js:25` to check the IP PeerTube detects?

**Comment by gbomacfly** [comment]
it gave me the ip plus the port. I forgot mto mention, that the port is also showing in the admin debug page.

`87.156.97.69:49162`

**Comment by Chocobozzz** [comment]
Your IP should not contain the port. Not sure which service in your config adds it though (IIS?)

**Comment by gbomacfly** [comment]
OK, thank you for the hint, I try to investigate a bit more.

**Comment by gbomacfly** [comment]
OK, update:

The port came from the Proxysettings in the ARR-Plugin (Applicatiopn Request Routing). Had to uncheck "Include TCP Port for client ip". Now there is no more port.

I changed the webserver section to:

```
webserver:
  https: false
  hostname: 'mytube.domain.de'
  port: 80 
```

No errormsg at refresh, correct IP in admin debug page, but upload from local disk fails with an "Unknow Error" and nothing in the logs.

To get rid of this, I had to change `preserveHostHeader` to `True` at system.webServer/proxy in the IIS-Configuration.

Now everything works fine, thank you! :)

**Comment by gbomacfly** [comment]
Juat a little update:
in order to allow mixed content (http/https), I had to add a line in the html-header:

`<meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests">`

Perhaps you can add this in an update?

Thanks again :)

**Comment by kanoriano** [comment]
Same problem. Main proxy machine IIS server (ip 10.0.0.10) with reverse proxy , peertube machine on hyper-v with debian (ip 10.0.0.24), im use standart installation instructions. Just input my dns settings and passwords.

Usual ARR settings on ISS:

![arr](https://github.com/user-attachments/assets/23ad50d9-1f83-4102-8b62-974e3645d4f4)

Backward proxy setup:

![rule](https://github.com/user-attachments/assets/4fd90994-4eb0-48b9-ae2f-30708372b3c9)

My settings:

listen:
  hostname: '127.0.0.1'
  port: 9000
webserver:
  https: true
  hostname: 'peertube.disona.live'
  port: 443
trust_proxy:
  - 'loopback'

At start or reload page i see error:

![err](https://github.com/user-attachments/assets/6e6d84aa-bd91-4085-a691-c00c7ec7b1d7)

Login error: 

![log](https://github.com/user-attachments/assets/ad52af15-fca9-4e8a-a572-b4ff8ddf2653)

I cant login until change hostname: 'peertube.disona.live' to hostname: '10.0.0.24' in production.yaml

Also trusted proxy errors.. Connection from pc on subnet:
 
![trp](https://github.com/user-attachments/assets/5db6755c-df14-4c0c-815b-69a7c9d5939e)

I try to use mobile phone with external network, to test connection and my ip fixed 10.0.0.10

If i put 2 strings below to nginx settings, my phone from external network show ip currectly (185.XXX.XXX.XXX). 

        set_real_ip_from 10.0.0.10;
	real_ip_header X-Forwarded-For;

But still errors with metric, same trusted proxy errors.

Also i can't watch and upload any content from external network. Show 0% until get Unknow Error. 

Only in my subnet, example pc with ip 10.0.0.55 i have success with upload and watching video.
Im really dont know how to fix it.
